### DISCUSSION TITLE
Custom choices in FormBuiler

### DISCUSSION THREAD

**Discussion by musaffarpatel** [PrestaShop/PrestaShop#34279] [Q&A] [answered]
The symfony form examples are great,  but in the following file:

demosymfonyformsimple/src/Form/DemoConfigurationFormType.php

I want to add a choice form control which itself is not difficult.  However, the options won't be hard coded but taken from a data source based on the ID in the url. 

Is it possible to pass custom choices to the form builder?

**Comment by kpodemski** [comment, marked as answer]
You can create and use data providers, one of the examples is `src/PrestaShopBundle/Form/Admin/Product/ProductPrice.php`, where choices are injected
